packages/xo-web/src/common/select-objects.js

```diff
@@ -372,10 +372,14 @@ const getPredicate = (state, props) => props.predicate
 
 export const SelectHost = makeStoreSelect(
   () => {
-    const getHostsByPool = createGetObjectsOfType('host')
-      .filter(getPredicate)
-      .sort()
-
+    const getHostsByPool = createSelector(
+      createGetObjectsOfType('host')
+        .filter(getPredicate)
+        .sort(),
+      (_, props) => props.pool,
+      (hosts, pool) =>
+        pool !== undefined ? hosts.sort(host => host.$pool !== pool) : hosts
+    )
```

Comment:
I'm not sure about specifically changing `SelectHost` to match your *exact* use case. Maybe we should expose a `compareOptions` and a `compareContainers` props on `GenericSelect` which would handle the sorting of options and containers. Then you'd simply have to pass
```js
compareContainers={(pool1, pool2) =>
  pool1.id === poolId ? -1 : pool2.id === poolId ? 1 : 0
}
```
to `SelectHost`.